### DISCUSSION TITLE
Debugging a RCL in Visual Studio

### DISCUSSION THREAD

**Discussion by cmaart** [RonSijm/RonSijm.Blazyload#2] [General]
Hi! 

For me, breakpoints are not loaded in lazy loaded RCLs, is that just my VS (config) or is this simply not possible atm? 
Not really related to the library but I thought just asking around here could make sense

![2024-04-04 18_00_03-WKO BlazorRCL](https://github.com/RonSijm/RonSijm.Blazyload/assets/4008452/7093b62e-844f-45f4-99a2-a4d0db0700eb)

**Comment by RonSijm** [comment]
I've had lots of issues with the debugger in the last couple versions of Visual Studio - Like the debugger window not showing up after hovering over values while hitting a breakpoint, and that kinda stuff - I don't know what the Visual Studio team has been doing to the debugger lately.

Debugging in Blazor has always been notoriously bad. It helps to not put code into the razor files, but create another partial class next to your component, and put all the code in there. I'd suggest you try that

**Reply by cmaart** [reply]
Yeah, it really is bad unfortunately :/
I tried setting a breakpoint in any class from the RCL but no breakpoint gets loaded (from your example WeatherLib1 for example tried `FetchData.razor` and also `WeatherResolver.cs`. 

Sad really because using lazy loaded RCLs would be a good way for us to build up a sort-of independent micro-frontend structure but of course debugging is key :/

**Comment by RonSijm** [comment]
I think I fixed it:
![image](https://github.com/RonSijm/RonSijm.Blazyload/assets/337928/474ac16a-213c-4483-b416-cc0d38630d3c)

There's a slight caveat though, it doesn't work yet immediately after the dll is loaded.

So the flow here was:
 - Set breakpoint in `WeatherLib2` / `OnInitializedAsync`
 - Run Application
 - Click `FD 2 - No Interface`
   - Dll and symbols get loaded, breakpoint doesn't hit yet
- Click Other item `FD 3 - Custom Class` for example
- Click `FD 2 - No Interface`
  - Dll and are already loaded, breakpoint hits

It's not ideal that it doesn't break the first time, but it's better than nothing.

I tried some other things, like adding the output directory as symbol source for the project, but that didn't have any effect either

**Reply by christoph-martin** [reply]
@RonSijm Okay, at the moment in my own example I only have 1 lazy loaded module unfortunately, so can't switch between 2 modules :/ This debugging experience of VS is really horrible for Blazor atm :D

**Reply by RonSijm** [reply]
I don't think you have to switch to a different module, it's just that the action that's causing the library to load (OnInitializedAsync) won't hit the first time.

I didn't explicitly test it, but I'd guess if you put a button on one of those pages with some action, and you put a breakpoint in the action, the breakpoint should hit. It's just the OnInitializedAsync, and maybe some others of the initial page load lifecycle

**Reply by christoph-martin** [reply]
@RonSijm  Ah okay, that doesn't work for me then :/ Simply won't enable my breakpoints in a RCL

**Reply by christoph-martin** [reply]
@RonSijm But something's off here for me anyways. Also this code in a RCL doesn't do anything when clicking the button but works in the host dll:
![image](https://github.com/RonSijm/RonSijm.Blazyload/assets/168006392/83d4de73-a05e-4184-9fa2-31ccadf87c92)

Pretty weird, initial render works but no interaction possible?!

EDIT: Only doesn't work in Edge, but works in Chrome. But hitting breakpoints is also not working in Chrome. Jesus, as a FE developer using React or Angular that's a horrible development experience
